### PR TITLE
Upgrade Ruby & Rails (Mini LIMS)

### PR DESCRIPTION
**Pull request by KatyTaylor**
N.B. I'm also planning to run a `bundle exec rake db:schema:dump` to make sure the schema file is correct for the Rails version. I will do this as a separate pull request after merging this and deploying it to training - because there are some differences between my local database and the training / prod ones, so running it locally won't produce the output we want.

- coffee-rails and web-console upgrades were necessary to resolve dependency conflicts (update: removed coffee-rails as we're not actually using coffee script)
- removal of reference to  was necessary as that method no longer exists in later Rails versions
- see https://stackoverflow.com/questions/37464966/what-causes-deprecation-warning-activerecordbase-raise-in-transactional-callb

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -32,20 +32,20 @@ gem 'exception_notification'
 # Use Capistrano for deployment
 # gem 'capistrano-rails', group: :development
 
-gem 'coveralls', require: false
+# gem 'coveralls', require: false
```
Comment (stevieing) [inline Gemfile:35]: No longer needed?

**Review by stevieing — APPROVED** [review]

File: app/models/batch/updater.rb [hunk 2]
```diff
@@ -19,7 +19,7 @@ def update!
     private
 
     def asset_params
-      { study: study, project: project, workflow_id: workflow, pipeline_destination_id: pipeline_destination,
+      { study: study, project: project, workflow_id: workflow.id, pipeline_destination_id: pipeline_destination,
```
Comment (JamesGlover) [inline app/models/batch/updater.rb:22]: I'd suggest being consistent and just using the object association across the board here, although rails usually does coerce a record to an id when passed directly to an id attribute (As it does for the other parameters in this, I'm not sure why you needed to be explicit for workflow)

File: config/application.rb [hunk 3]
```diff
@@ -23,8 +23,11 @@ class Application < Rails::Application
     config.mailer = YAML.load_file("#{Rails.root}/config/mailer.yml")[Rails.env]
 
     # Do not swallow errors in after_commit/after_rollback callbacks.
```
Comment (JamesGlover) [inline config/application.rb:25]: Remove this comment, as it relates to the removed config option

**Review by JamesGlover — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 1 above)
Comment (KatyTaylor) [reply Gemfile]: Yep, replaced by using simplecov directly. I've removed the comment now.

File: app/controllers/assets_controller.rb [hunk 4]
```diff
@@ -34,7 +34,7 @@ def search
   end
 
   def assets_provided
-    params[:assets].is_a?(Hash) && params[:assets].keys.present?
+    params[:assets].is_a?(ActionController::Parameters) && params[:assets].keys.present?
```
[the remark this answers is not in the document]
Comment (KatyTaylor) [reply app/controllers/assets_controller.rb:37]: OK sure thanks. It doesn't check the type, but that seems unnecessary anyway.

**Review by JamesGlover — APPROVED** [review]

**Pull request merged**